File: app/src/main/java/com/app_investigarte/WelcomeActivity.java
package com.app_investigarte;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class WelcomeActivity extends AppCompatActivity {
    public Button btnStart;
    public Intent intent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome);

        SharedPreferences preferencia = getSharedPreferences(getString(R.string.PREFERENS), Context.MODE_PRIVATE);

        Bundle putExtra = this.getIntent().getExtras();
        String userName = putExtra.getString("username");

        preferencia.edit().putString("NAME_USER", userName).apply();

        TextView user = findViewById(R.id.txtusername);
        user.setText(userName);

        //pasar de activity al precionar el boton.
        btnStart = findViewById(R.id.btnStart);
        btnStart.setOnClickListener(view -> {
            intent = new Intent(this, NavDrawerActivity.class);
            startActivity(intent);
            super.finishAffinity();
            // intent.putExtra("user",MainActivity.this.getUser());
            // Toast.makeText(MainActivity.this,"Bienvenido"+inputUser,Toast.LENGTH_LONG).show();
        });
    }
}

File: app/src/main/java/com/app_investigarte/WelcomeActivity.kt
package com.app_investigarte

import android.content.Intent
import android.media.MediaPlayer
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.app_investigarte.databinding.ActivityWelcomeBinding


class WelcomeActivity : AppCompatActivity() {


    //implementar ViewBinding en activity
    private lateinit var binding: ActivityWelcomeBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityWelcomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //ago una intacion del sonido pop.mp3
        val pop = MediaPlayer.create(this, R.raw.pop)

        //se recuperan los datos de usuario logeado
        val pref = getSharedPreferences(getString(R.string.PREFERENS), MODE_PRIVATE)
        val email = pref.getString("email","")
        val name = pref.getString("name", "")
        val provider = pref.getString("provider", "")

        if (provider != null) {
            if(provider.equals(ProviderType.GOOGLE.toString())){
                binding.txtusername.text = email
            }else{
                binding.txtusername.text = name
            }
        }



        //pasar de activity al precionar el boton.
        binding.btnStart.setOnClickListener { view: View? ->
            pop.start()
            startActivity(Intent(this, NavDrawerActivity::class.java))
            overridePendingTransition(R.anim.slide_in,R.anim.slide_out)
            super.finishAffinity()
        }
    }
}